libcloudforensics/providers/aws/forensics.py

```diff
@@ -268,6 +275,8 @@ def StartAnalysisVm(
     ami = ami_list[0]['ImageId']
   assert ami  # Mypy: assert that ami is not None
 
+  userdata = utils.ReadStartupScript(userdata_file)
```

Comment:
After you've changed `userdata_file: Optional[str] = utils.FORENSICS_STARTUP_SCRIPT_AWS` to `userdata_file: Optional[str] = None`, you can add above this line:
```
if not userdata_file:
  userdata_file = utils.FORENSICS_STARTUP_SCRIPT_AWS
```

The reason is that currently in`aws_cli.py` you are passing `args.launch_script` to the function directly, which defaults to `None` and will thus override `userdata_file: Optional[str] = utils.FORENSICS_STARTUP_SCRIPT_AWS`. This will cause `utils.ReadStartupScript()` to be called with `None` which in turn will cause the wrong script to be loaded (`STARTUP_SCRIPT` instead of `FORENSICS_STARTUP_SCRIPT_AWS`)

Reply:
Done. I tested that out, but only that it got _a_ script /facepalm